src/types/stardust/block/output/mod.rs

```diff
@@ -152,6 +153,12 @@ impl Output {
             Self::Foundry(_) => true,
         }
     }
+
+    pub fn raw(self) -> Result<Vec<u8>, bee_block_stardust::Error> {
+        let ctx = bee_block_stardust::protocol::protocol_parameters();
+        let bee_output = bee_block_stardust::output::Output::try_from_with_context(&ctx, self)?;
+        Ok(bee_output.pack_to_vec())
+    }
```

Comment:
`bee_block_stardust::protocol::protocol_parameters()` represents the Shimmer protocol parameters and should mainly be used for testing. This is very fragile (it sometimes does not work with the `private_tangle` for example). We should retrieve the proper context here, if context-free conversions are not available.

Reply:
I know that, but for some reason I need the token supply to being able to create bee output type. I haven't grasped that fully yet. Any suggestion?

Reply:
I look into another way ...

Reply:
Yeah, since the fields in `bee-block` are mainly private, we can't really create these types in an ad-hoc manner. The only option that we have right now is to get the protocol parameters that were active at the time the output was issued. We have been in talks with @thibault-martinez to allow such functionality directly in Bee.